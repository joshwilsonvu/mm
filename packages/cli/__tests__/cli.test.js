const execa = require("execa");
const cliPath = require.resolve("..");

/*
 * The best way to test the CLI's functionality is to test the application as a whole,
 * since it is responsible for building, serving, and reloading MagicMirror. Therefore,
 * tests in e2e.test.js may uncover problems here that these tests don't find.
 */
async function testOutput(...args) {
  const result = await execa(cliPath, args);
  expect(result.stdout).toMatchSnapshot();
}
test("shows help message on '$ mm'", () => testOutput());

test("shows help message on '$ mm start -h'", () => testOutput("start", "--help"));

test("shows help message on '$ mm build -h'", () => testOutput("build", "--help"));

test("shows help message on '$ mm serve -h'", () => testOutput("serve", "--help"));