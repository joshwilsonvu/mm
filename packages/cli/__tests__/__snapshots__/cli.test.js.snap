// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows help message on '$ mm build -h' 1`] = `
"[log] mm build

Create an optimized build

Options:
  --cwd        run mm in this directory                                 [string]
  --config     the path to the MagicMirror config file                  [string]
  --log-level  the minimum log level to be shown
                     [choices: \\"fatal\\", \\"error\\", \\"warn\\", \\"log\\", \\"info\\", \\"debug\\"]
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --analyze    after the build, show the contents of the bundle in a web page"
`;

exports[`shows help message on '$ mm serve -h' 1`] = `
"[log] mm serve

Run MagicMirror from a build

Options:
  --cwd        run mm in this directory                                 [string]
  --config     the path to the MagicMirror config file                  [string]
  --log-level  the minimum log level to be shown
                     [choices: \\"fatal\\", \\"error\\", \\"warn\\", \\"log\\", \\"info\\", \\"debug\\"]
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]
  --build      if MagicMirror hasn't been built, run a build before serving
  --browser    open a web browser"
`;

exports[`shows help message on '$ mm start -h' 1`] = `
"[log] mm start

Start serving MagicMirror in development mode

Options:
  --cwd         run mm in this directory                                [string]
  --config      the path to the MagicMirror config file                 [string]
  --log-level   the minimum log level to be shown
                     [choices: \\"fatal\\", \\"error\\", \\"warn\\", \\"log\\", \\"info\\", \\"debug\\"]
  --help        Show help                                              [boolean]
  --version     Show version number                                    [boolean]
  --browser     open a web browser instead of an electron window
  --serveronly  don't open an electron window or a web browser"
`;

exports[`shows help message on '$ mm' 1`] = `
"[log] mm [command]

Commands:
  mm start  Start serving MagicMirror in development mode
  mm build  Create an optimized build
  mm serve  Run MagicMirror from a build

Options:
  --cwd        run mm in this directory                                 [string]
  --config     the path to the MagicMirror config file                  [string]
  --log-level  the minimum log level to be shown
                     [choices: \\"fatal\\", \\"error\\", \\"warn\\", \\"log\\", \\"info\\", \\"debug\\"]
  --help       Show help                                               [boolean]
  --version    Show version number                                     [boolean]

Run mm <command> --help for more informaton about each command."
`;
